modules/release-notes/ref-release-notes-compatibility-matrix.adoc

```diff
@@ -0,0 +1,41 @@
+:_content-type: REFERENCE
+[id="compatibility-matrix"]
+= Compatibility matrix
+
+The following table lists products and their versions compatible with {product-short} {product-version}:
+
+.Platform Compatibility Matrix
+[cols="1,2,1", options="header"]
+|===
+| {product-short} | Platform | {product-custom-resource-type} Version
+
+| `{product-version}`
+|
+- {ocp-brand-name} `{ocp-version-min}`-`{ocp-version}`
+
+- {eks-brand-name} ({eks-short})
+
+- {aks-brand-name} ({aks-short})
+
+- {aro-brand-name} ({aro-short})
+
+- {rosa-brand-name} ({rosa-short})
+
+- {gke-brand-name} ({gke-short})
+| `{product-backstage-version}`
+| `1.3`
```

Comment:
Do we want/need a 1.3 table in the 1.4 docs? Or should this be added instead in the `release-1.3` branch and published to the 1.3 docs?